### ISSUE TITLE
load two swagger generated APIs in angular 5 / Typscript project

### ISSUE THREAD

**Issue by AlexanderGoerlich** [swagger-api/swagger-codegen#7354]
##### Description

We have exported with Swagger 2 APIS: 

1. typescript-angular-client-generated.zip (UsersService)
2. typescript-angular-client-generated.zip (DataService)

API-1: http://example-myapi.de:8081/v2 UsersService
API-2: http://example-myapi.de:8082/v2 DataService

We try to import that...but we can't load 2 api ...UsersService and DataService.

`
 constructor(
    private usersservice: UsersService,
    private dataservice: DataService
){}
`
no Error when load one 
`
 constructor(
    private usersservice: UsersService,
    // private dataservice: DataService
){}
`
or `
 constructor(
    // private usersservice: UsersService,
    private dataservice: DataService
){}
`

##### Error message

ERROR Error: Uncaught (in promise): Error: StaticInjectorError[DataService]: 
  StaticInjectorError[DataService]: 
    NullInjectorError: No provider for DataService!
Error: StaticInjectorError[DataService]: 
  StaticInjectorError[DataService]: 
    NullInjectorError: No provider for DataService!

##### Swagger-codegen version

2.3.0 with language typescript-angular

**How can I use two apis exports in one project. Thx a lot for helping**

#####  Swagger files

[typescript-angular-client-generated(UserService).zip](https://github.com/swagger-api/swagger-codegen/files/1617917/typescript-angular-client-generated.UserService.zip)
[typescript-angular-client-generated(DataService).zip](https://github.com/swagger-api/swagger-codegen/files/1617918/typescript-angular-client-generated.DataService.zip)

**Comment by macjohnny** [comment]
@AlexanderGoerlich 
The error states that some providers of services are missing, which is the case when you don't import the `ApiModule`.
Since you are generating two separate APIs and therefore need to import both `ApiModule` modules that are generated. To do so in one single `app.module.ts`, you can create an alias of the imported name to avoid naming conflicts:
```
import { ApiModule } from 'my-api-path';
import { ApiModule as OtherApiModule } from 'my-other-api-path';

@NgModule({
  imports: [
    ApiModule, 
    OtherApiModule
  ]
})
export class CoreModule {

}
```
See https://github.com/swagger-api/swagger-codegen/issues/6982#issuecomment-344952458

**Comment by AlexanderGoerlich** [comment]
@macjohnny **thy a lot** 👍 

does it make sense to add here a short docu?

https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/petstore/typescript-angular-v4.3/npm

Should I write a short comment on that file an make a PR?

**Comment by macjohnny** [comment]
@AlexanderGoerlich I am preparing a PR to improve the docs

**Comment by macjohnny** [comment]
@wing328 @AlexanderGoerlich this issue can be closed, as the doc improvements were merged in #7363
